### PR TITLE
fix(analytics-chart): keep line borderwidth single dimension [MA-2547]

### PR DESCRIPTION
**Pull request by mihai-peteu**
# Summary

fix for https://kongstrong.slack.com/archives/C03NQNBSSNL/p1706294909888809?thread_ts=1706242086.056759&cid=C03NQNBSSNL

https://konghq.atlassian.net/browse/MA-2547

### CODE REVIEW

File: packages/analytics/analytics-chart/src/components/AnalyticsChart.vue [hunk 1]
```diff
@@ -202,6 +202,7 @@ const computedChartData = computed(() => {
   return isTimeSeriesChart.value
     ? composables.useExploreResultToTimeDataset(
       {
+        borderWidth: 2,
```
Comment (filipgutica) [inline packages/analytics/analytics-chart/src/components/AnalyticsChart.vue:205]: Remove

File: packages/analytics/analytics-chart/src/types/explore-to-dataset-deps.ts [hunk 2]
```diff
@@ -4,13 +4,16 @@ import type { AnalyticsChartColors } from './chart-data'
...
+ * borderWidth
+ *   - determines the line chart line thickness; defaults to zero
  * fill
  *   - determines if the generated should have the area under the line filled
  * useMetricForColor
  *   - determines if colors should be generated based on the metrics
  *   - if false, colors will be auto generated using the generic datavis palette
  */
 export interface ExploreToDatasetDeps {
+  borderWidth?: number
```
Comment (filipgutica) [inline packages/analytics/analytics-chart/src/types/explore-to-dataset-deps.ts:16]: Remove

File: packages/analytics/analytics-chart/src/composables/useExploreResultToTimeDatasets.ts [hunk 3]
```diff
@@ -176,6 +176,7 @@ export default function useExploreResultToTimeDataset(
             total: filled.reduce((acc, { y }) => acc + Number(y), 0),
             ...defaultLineOptions,
             fill,
+            borderWidth: borderWidth && !fill, // only show chart line if single dimension, and not "filled"
```
Comment (filipgutica) [inline packages/analytics/analytics-chart/src/composables/useExploreResultToTimeDatasets.ts:179]: The rule should be show border if not filled. Dimensionality should not be used to determine it

**Review by filipgutica — CHANGES_REQUESTED** [review]

**Comment by mihai-peteu** [comment]
Closing in favor of https://github.com/Kong/public-ui-components/pull/1115

**Pull request closed without merging**